File: Guanabara - PHP/exercicios-PHPOO/Aula12-Polimorfismo/Ave.php
<?php
include_once 'Animal.php';
class Ave extends Animal
{
    //Atributo
    protected $corPena;
    
    //Método Especial
    public function fazerNinho(){
        echo "<p>Fazendo ninho...</p>";
    }
    
    //Métodos Implementados usando polimorfismo de sobreposição
    public function alimentar() {
        echo '<p>Ave se alimentando...</p>';
    }
    public function emitirSom() {
        echo '<p>Ave piando...</p>';
    }
    public function locomover() {
        echo '<p>Ave voando...</p>';
    }
    
    //Métodos Acessores
    public function getCorPena() {
        return $this->corPena;
    }

    public function setCorPena($corPena): void {
        $this->corPena = $corPena;
    }
}

File: Java/Meus exercícios POO/Aula12.Polimorfismo/src/main/java/polimorfismo/Ave.java
package polimorfismo;
public class Ave extends Animal { //tipo de herança: 'para diferença'
    //Atributo
    private String corPena;
    
    //Método Público
    public void fazerNinho() {
        System.out.println("Construindo ninho...");
    }
    
    //Implementando Métodos Abstratos (Polimorfismo de Sobreposição)
    @Override
    public void locomover() {
        System.out.println("Voando...");
    }

    @Override
    public void alimentar() {
        System.out.println("Comendo frutas...");
    }

    @Override
    public void emitirSom() {
        System.out.println("Piando...");
    }
    
    //Métodos Acessores
    public String getCorPena() {
        return corPena;
    }

    public void setCorPena(String corPena) {
        this.corPena = corPena;
    }
}
